lib/api/redfish-1.0/chassis.js

```diff
@@ -286,36 +311,82 @@ var getThermal = controller(function(req, res) {
     var systemId = req.swagger.params.identifier.value.split('.')[1];
     var options = redfish.makeOptions(req, res, req.swagger.params.identifier.value);
 
-    return dataFactory(chassisId, 'system')
-    .then(function(system) {
-        return Promise.all([
-            dataFactory(systemId || system, 'sdrData'),
-            waterline.nodes.findByIdentifier(systemId || system)
-            .then(function(obj) {
-                return env.get('config.redfish', '{}', obj && obj.sku ? [ obj.sku ] : null );
+    return redfish.getVendorNameById(chassisId)
+    .then(function(result) {
+        if (result.vendor === 'Cisco') {
+            return dataFactory(chassisId, 'ucsThermal')
+            .then(function(thermalData) {
+                var ucsPowerThermalData = thermalData.ucsPowerThermalData;
+                var ucsFanData = thermalData.ucsFanData;
+                var regxObj = /.*_temp_sen_.*[^avg|max|min]$/;
+                options.tempList = [];
+                options.fanList = [];
+                _.forEach(ucsPowerThermalData, function(data) {
+                    _.forEach(data, function(item) {
+                        _.forEach(_.keys(item), function(key) {
+                            if(regxObj.test(key)) {
+                                options.tempList.push(
+                                    {
+                                        'sensorId': key.match(regxObj)[0],
+                                        'sensorReading': item.key
+                                    }
+                                );
+                            }
+                        });
+                    });
+                });
+                _.forEach(ucsFanData, function(data) {
+                    _.forEach(data, function(item) {
+                        var sensorId = item.dn.split('/').slice(0, -1).join('_');
+                        options.fanList.push(
+                            {
+                                'sensorId': sensorId,
+                                'sensorReading': item.speed
+                            }
+                        );
+                    });
+                });
+                return redfish.render('redfish.1.0.0.thermal.1.0.0.json',
```

Comment:
If you are using the same redfish.render with other platforms, you can combine following 6 lines.